### PR TITLE
Switch project on VueJS

### PR DESCRIPTION
**Pull request by Quetzacoalt91**
This PR reworks the interface in order to run with VueJS.
Working with Machine Shuffle, it allows users to test PrestaShop on a dedicated shop.

### CODE REVIEW

File: public/index.html [hunk 1]
```diff
@@ -0,0 +1,14 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <meta charset="utf-8">
+    <meta name="viewport" content="width=device-width,initial-scale=1.0">
+    <title>PrestaShop Live Demo</title>
+    <meta name="description" content="Looking for a powerful shopping cart software? Try the PrestaShop Demo now and visualize it on mobile, tablet and desktop." />
+		<link href="https://fonts.googleapis.com/icon?family=Material+Icons" rel="stylesheet">
```
Comment (PierreRambaud) [inline public/index.html]: mix between spaces and tabs

File: src/App.vue [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+export default {
+  name: 'App',
+  components: {
+    Header,
+    HeaderSwitch,
+  },
+  data() {
+    return {
+      device: 'desktop',
```
Comment (PierreRambaud) [inline src/App.vue]: is this data used?

File: src/components/DeviceButton.vue [hunk 3]
```diff
@@ -0,0 +1,30 @@
+<template>
+  <li>
+    <a class='change-device'
```
Comment (PierreRambaud) [inline src/components/DeviceButton.vue]:
```suggestion
    <a class="change-device"
```

File: src/components/HeaderSwitch.vue [hunk 4]
```diff
@@ -0,0 +1,25 @@
+<template>
+  <a class="btn btn-collapse"
+      v-on:click="toggleHeader()"
+      :class="[!displayHeader ? 'collapsed': '']"
+  >
+    <span class="hide-header" v-show="displayHeader">{{ $t("hide") }}</span>
```
Comment (PierreRambaud) [inline src/components/HeaderSwitch.vue]:
```suggestion
    <span class="hide-header" v-if="displayHeader">{{ $t("hide") }}</span>
```

File: src/components/HeaderSwitch.vue [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+    <span class="show-header" v-show="!displayHeader">{{ $t("show") }}</span>
```
Comment (PierreRambaud) [inline src/components/HeaderSwitch.vue]:
```suggestion
    <span class="show-header" v-else>{{ $t("show") }}</span>
```

**Comment by PierreRambaud** [comment]
Good job! :+1:

File: src/App.vue (hunk 2 above)
Comment (Quetzacoalt91) [reply src/App.vue]: Not anymore, components get this data from the store directly

File: src/store.js [hunk 6]
```diff
@@ -75,7 +75,7 @@ export default {
         // Make sure a webserver is started before updating the iframe,
         // to avoid error messages at startup
         const recurrentCheck = setInterval(() => {
-          Vue.http.head(`http:/${container.name}.${state.factory.baseEndpoint}`).then((headResponse) => {
+          Vue.http.head(`http:/${container.name}.${state.factory.baseEndpoint}/error500.html`).then((headResponse) => {
```
Comment (PierreRambaud) [inline src/store.js]: should ti be in https instead? (or configurable?)

Comment (Quetzacoalt91) [reply src/store.js]: Looks like we are unable to provide HTTPS for the shops in any environment, as their domain cannot be linked to the certificate we have (no wildcard)

**Comment by PierreRambaud** [comment]
Remove WIP when it's ready for review :)

File: src/components/Shop.vue [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+    setIframeReady() {
+      if (this.iframeUrl) {
+        this.$store.commit('setIframeReady');
+      }
+    },
+  },
+  created() {
+    const payload = {
+      idmodule: this.$route.query.idmodule,
+      moduleNameToinstall: this.$route.query.module_name_toinstall,
```
Comment (PierreRambaud) [inline src/components/Shop.vue]:
```suggestion
      moduleNameToInstall: this.$route.query.module_name_toinstall,
```

File: src/components/Shop.vue [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+    },
+    setIframeReady() {
+      if (this.iframeUrl) {
+        this.$store.commit('setIframeReady');
+      }
+    },
+  },
+  created() {
+    const payload = {
+      idmodule: this.$route.query.idmodule,
```
Comment (PierreRambaud) [inline src/components/Shop.vue]:
```suggestion
      idModule: this.$route.query.idmodule,
```

Comment (Quetzacoalt91) [reply src/components/Shop.vue]: That wouldn't be the only place I have to modify the case. As my current objective is to push to production the current version with SSL feature, I'd like to keep it as-is.

File: src/components/DeviceButton.vue [hunk 9]
```diff
@@ -0,0 +1,30 @@
+<template>
+  <li>
+    <a class="change-device"
+      :class="[
+        deviceTrigger,
+        device == deviceTrigger ? 'active': ''
+      ]"
+      v-on:click="displayOn(deviceTrigger)"
```
Comment (NeOMakinG) [inline src/components/DeviceButton.vue]: Next in the project, you use some directives with shorthand practice, Vue recommend to use one or the other (shorthand or complete directive), I would then recommend you to use @click or v-on everywhere but not mix them in the entire project!

Comment (Quetzacoalt91) [reply src/components/DeviceButton.vue:8]: I created https://github.com/PrestaShop/live-demo-devices/issues/15, so we can move forward with this PR without forgetting your comment.

**Pull request merged**